kem/mceliece/gen.go

```diff
@@ -0,0 +1,162 @@
+//go:build ignore
+// +build ignore
+
+// Autogenerates wrappers from templates to prevent too much duplicated code
+// between the code for different modes.
+package main
+
+import (
+	"bytes"
+	"go/format"
+	"io/ioutil"
+	"strings"
+	"text/template"
+)
+
+type Param struct {
+	Gf             string
+	PublicKeySize  uint
+	PrivateKeySize uint
+	CiphertextSize uint
+	SysN           uint
+	SysT           uint
+}
+
+type Instance struct {
+	Name  string
+	Param Param
+}
+
+func (m Instance) Pkg() string {
+	return strings.ToLower(m.Name)
+}
+
+func (m Instance) IsSemiSystematic() bool {
+	return strings.HasSuffix(m.Name, "f")
+}
+
+func (m Instance) Is348864() bool {
+	return strings.Contains(m.Name, "348864")
+}
+
+func (m Instance) Is460896() bool {
+	return strings.Contains(m.Name, "460896")
+}
+
+func (m Instance) Is6688128() bool {
+	return strings.Contains(m.Name, "6688128")
+}
+
+func (m Instance) Is6960119() bool {
+	return strings.Contains(m.Name, "6960119")
+}
+
+func (m Instance) Is8192128() bool {
+	return strings.Contains(m.Name, "8192128")
+}
+
+var (
+	McElieceParam348864 = Param{
+		Gf:             "gf2e12",
+		PublicKeySize:  261120,
+		PrivateKeySize: 6492,
+		CiphertextSize: 96,
+		SysN:           3488,
+		SysT:           64,
+	}
+	McElieceParam460896 = Param{
+		Gf:             "gf2e13",
+		PublicKeySize:  524160,
+		PrivateKeySize: 13608,
+		CiphertextSize: 156,
+		SysN:           4608,
+		SysT:           96,
+	}
+	McElieceParam6688128 = Param{
+		Gf:             "gf2e13",
+		PublicKeySize:  1044992,
+		PrivateKeySize: 13932,
+		CiphertextSize: 208,
+		SysN:           6688,
+		SysT:           128,
+	}
+	McElieceParam6960119 = Param{
+		Gf:             "gf2e13",
+		PublicKeySize:  1047319,
+		PrivateKeySize: 13948,
+		CiphertextSize: 194,
+		SysN:           6960,
+		SysT:           119,
+	}
+	McElieceParam8192128 = Param{
+		Gf:             "gf2e13",
+		PublicKeySize:  1357824,
+		PrivateKeySize: 14120,
+		CiphertextSize: 208,
+		SysN:           8192,
+		SysT:           128,
+	}
+	Instances = []Instance{
+		{Name: "mceliece348864", Param: McElieceParam348864},
```

Comment:
I propose to use names in line with liboqs here:
https://github.com/open-quantum-safe/liboqs/blob/6e1f49aa48971b0fc8658e94ca41d1ad4fe480ee/src/kem/kem.h#L42

This also matches the capitalization of e.g. `Kyber512`